trace/sql/integration_test.go

```diff
@@ -6,92 +6,102 @@ import (
 )
 
-type SQLTestSuite struct {
-	suite.Suite
+type dockerRuntime struct {
 	sql  *dockertest.Resource
 	pool *dockertest.Pool
-	db   *DB
-	mtr  *mocktracer.MockTracer
-	ctx  context.Context
 }
 
-func TestSQLTestSuite(t *testing.T) {
-	suite.Run(t, new(SQLTestSuite))
+func TestMain(m *testing.M) {
+
+	d := dockerRuntime{}
+
+	err := startUpContainerSync(&d)
+	if err != nil {
+		log.Errorf("could not start containers %v", err)
+		os.Exit(1)
+	}
+
+	exitVal := m.Run()
+
+	err = tearDownContainerSync(&d)
+	if err != nil {
+		log.Errorf("could not tear down containers %v", err)
+		os.Exit(1)
+	}
+
+	os.Exit(exitVal)
 }
 
-func (s *SQLTestSuite) SetupSuite() {
-	s.ctx = context.Background()
+func startUpContainerSync(d *dockerRuntime) error {
 
 	pool, err := dockertest.NewPool("")
-	s.NoError(err)
-	s.pool = pool
-	s.pool.MaxWait = time.Minute * 2
+	if err != nil {
+		return err
+	}
+	d.pool = pool
+	d.pool.MaxWait = time.Minute * 2
 
-	s.sql, err = s.pool.RunWithOptions(&dockertest.RunOptions{Repository: "mysql",
+	d.sql, err = d.pool.RunWithOptions(&dockertest.RunOptions{Repository: "mysql",
 		Tag: "5.7.25",
 		PortBindings: map[docker.Port][]docker.PortBinding{
 			"3306/tcp":  {{HostIP: "", HostPort: "3307"}},
 			"33060/tcp": {{HostIP: "", HostPort: "33061"}},
 		},
 		ExposedPorts: []string{"3306/tcp", "33060/tcp"},
 		Env: []string{
-			fmt.Sprintf("MYSQL_ROOT_PASSWORD=%s", DBRootPassword),
-			fmt.Sprintf("MYSQL_USER=%s", DBUsername),
-			fmt.Sprintf("MYSQL_PASSWORD=%s", DBPassword),
-			fmt.Sprintf("MYSQL_DATABASE=%s", DBSchema),
+			fmt.Sprintf("MYSQL_ROOT_PASSWORD=%s", dbRootPassword),
+			fmt.Sprintf("MYSQL_USER=%s", dbUsername),
+			fmt.Sprintf("MYSQL_PASSWORD=%s", dbPassword),
+			fmt.Sprintf("MYSQL_DATABASE=%s", dbSchema),
 			"TIMEZONE=UTC",
 		}})
-	s.NoError(err)
+	if err != nil {
+		return err
+	}
 
-	// optional : enable logs for the container
-	//s.TailLogs(s.sql.Container.ID, os.Stdout)
+	// optionally print the container logs in stdout
+	//TailLogs(d, d.sql.Container.ID, os.Stdout)
 
 	// wait until the container is ready
-	err = s.pool.Retry(func() error {
-		connString := fmt.Sprintf("%s:%s@(%s:%s)/%s?parseTime=true&multiStatements=true", DBUsername, DBPassword, DBHost, DBPort, DBSchema)
-		db, err := Open("mysql", connString)
+	err = d.pool.Retry(func() error {
+		db, err := sqlx.Open("mysql", fmt.Sprintf(connectionFormat, dbUsername, dbPassword, dbHost, dbPort, dbSchema))
 		if err != nil {
 			// container not ready ... return error to try again
 			return err
 		}
-		s.NotNil(db)
-		s.db = db
-		return db.Ping(s.ctx)
+		return db.Ping()
 	})
-	s.NoError(err)
-
 	// start up any other services
-
-	s.mtr = mocktracer.New()
-	opentracing.SetGlobalTracer(s.mtr)
-
-	s.db.SetConnMaxLifetime(time.Minute)
-	s.db.SetMaxIdleConns(10)
-	s.db.SetMaxOpenConns(10)
-
+	return nil
 }
 
-func (s *SQLTestSuite) TailLogs(containerID string, out io.Writer) {
+func TailLogs(d *dockerRuntime, containerID string, out io.Writer) {
```

Comment:
as the linter suggests we can remove this.

Reply:
i ve just enabled it. So now we see the logs from the container. Would that be ok ?